### PR TITLE
Enhance Job CLI to support multi-apps/sites 

### PR DESCRIPTION
**Pull request by chesterxgchen**
### Description
The existing CLI Job only support simple app, i.e. all client's configuration and code the same. When change the configurations, all sites/clients configurations are changed. This PR enhanced the command to enable this feature. 

Specifically,  if user needs create an job folder using multi-site configurations such as 
```
server/config/config_fed_server.conf
site-1/config/config_fed_client.conf
site-2/config/config_fed_client.conf
```
user can now need to use the following command to change the site-specific configuratioins

```
nvflare job create \
 -j /tmp/nvflare/my_job \
-w sag_pt_deploy_map \
-f server/config_fed_server.conf min_clients=2 num_rounds=2 \
-f site-2/config_fed_client.conf app_script=train.py \
-f site-1/config_fed_client.conf app_script=train.py \
-s ~/projects/NVFlare/examples/hello-world/step-by-step/cifar10/code/fl/train.py \
-sd ~/projects/NVFlare/examples/hello-world/step-by-step/cifar10/code/fl 
-a site-1 k1=v1 k2=v2
-a site-2 k3=v3 k4=v3

```
Here is the example of output

```
 nvflare job create  -j /tmp/nvflare/my_job -w sag_pt_deploy_map -f server/config_fed_server.conf min_clients=2 num_rounds=2 -f site-2/config_fed_client.conf app_script=train.py -f site-1/config_fed_client.conf app_script=train.py  -s ~/projects/NVFlare/examples/hello-world/step-by-step/cifar10/code/fl/train.py -sd ~/projects/NVFlare/examples/hello-world/step-by-step/cifar10/code/fl -a site-1 k1=v1 k2=v2 -a site-2 k3=v3 k4=v3 -force

The following are the variables you can change in the template

---------------------------------------------------------------------------------------------------------------------------------------

  job folder: /tmp/nvflare/my_job

---------------------------------------------------------------------------------------------------------------------------------------
  file_name                      var_name                       value                               component
---------------------------------------------------------------------------------------------------------------------------------------
  meta.conf                      mandatory_clients              []
  meta.conf                      min_clients                    2
  meta.conf                      server                         ['server']
  meta.conf                      site-1                         ['site-1']
  meta.conf                      site-2                         ['site-2']

  app: site-1

  config_fed_client.conf         app_config                     --k1 v1 --k2 v2
  config_fed_client.conf         app_script                     train.py
  config_fed_client.conf         data_exchange_path
  config_fed_client.conf         global_evaluation              True
  config_fed_client.conf         heartbeat_interval             5.0
  config_fed_client.conf         heartbeat_timeout              60
  config_fed_client.conf         params_exchange_format         pytorch
  config_fed_client.conf         params_transfer_type           DIFF
  config_fed_client.conf         pipe_name                      pipe
  config_fed_client.conf         read_interval                  0.1
  config_fed_client.conf         result_poll_interval           0.1
  config_fed_client.conf         script                         python custom/{app_script}  {app_co SubprocessLauncher
  config_fed_client.conf         training                       True
  config_fed_client.conf         workers                        1

  app: server

  config_fed_server.conf         allow_empty_global_weights     False                               ScatterAndGather
  config_fed_server.conf         best_global_model_file_name    best_FL_global_model.pt             PTFileModelPersistor
  config_fed_server.conf         expected_data_kind             WEIGHT_DIFF                         InTimeAccumulateWeightedAggregator
  config_fed_server.conf         global_model_file_name         FL_global_model.pt                  PTFileModelPersistor
  config_fed_server.conf         ignore_result_error            False                               ScatterAndGather
  config_fed_server.conf         key_metric                     accuracy                            IntimeModelSelector
  config_fed_server.conf         min_clients                    2                                   ScatterAndGather
  config_fed_server.conf         negate_key_metric              False                               IntimeModelSelector
  config_fed_server.conf         num_rounds                     2                                   ScatterAndGather
  config_fed_server.conf         persist_every_n_rounds         1                                   ScatterAndGather
  config_fed_server.conf         snapshot_every_n_rounds        1                                   ScatterAndGather
  config_fed_server.conf         start_round                    0                                   ScatterAndGather
  config_fed_server.conf         task_check_period              0.5                                 ScatterAndGather
  config_fed_server.conf         train_timeout                  0                                   ScatterAndGather
  config_fed_server.conf         validation_metric_name         initial_metrics                     IntimeModelSelector
  config_fed_server.conf         wait_time_after_min_received   0                                   ScatterAndGather
  config_fed_server.conf         weigh_by_local_iter            False                               IntimeModelSelector

  app: site-2

  config_fed_client.conf         app_config                     --k3 v3 --k4 v3
  config_fed_client.conf         app_script                     train.py
  config_fed_client.conf         data_exchange_path
  config_fed_client.conf         global_evaluation              True
  config_fed_client.conf         heartbeat_interval             5.0
  config_fed_client.conf         heartbeat_timeout              60
  config_fed_client.conf         params_exchange_format         pytorch
  config_fed_client.conf         params_transfer_type           DIFF
  config_fed_client.conf         pipe_name                      pipe
  config_fed_client.conf         read_interval                  0.1
  config_fed_client.conf         result_poll_interval           0.1
  config_fed_client.conf         script                         python custom/{app_script}  {app_co SubprocessLauncher
  config_fed_client.conf         training                       True
  config_fed_client.conf         workers                        1

---------------------------------------------------------------------------------------------------------------------------------------
```

**Additional change**

-w --template option is now support both template_name or template_folder. 
you can specify 
    -w job_template/sag_pt  or -w sag_pt

For other job templates ( non-site specific job templates), the output and the commands are the same

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by IsaacYangSLA** [comment]
Does this PR support job submission with multi-apps/sites?

**Review by IsaacYangSLA — APPROVED** [review]
I think the index portion can be improved in the future because it has logic directly on filename/path.  Other than that, it looks good to me.

File: job_templates/sag_pt_deploy_map/info.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    description = "SAG workflow with pytorch, deploy_map, dsite-specific configs"
```
Comment (YuanTingHsieh) [inline job_templates/sag_pt_deploy_map/info.conf:2]: typo? "dsite-specific configs" -> "site-specific configs"?

File: job_templates/sag_pt_deploy_map/server/config_fed_server.conf [hunk 2]
```diff
@@ -0,0 +1,102 @@
+{
+  # version of the configuration
+  format_version = 2
+
+  # task data filter: if filters are provided, the filter will filter the data flow out of server to client.
+  task_data_filters =[]
+
+  # task result filter: if filters are provided, the filter will filter the result flow out of server to client.
+  task_result_filters = []
```
Comment (YuanTingHsieh) [inline job_templates/sag_pt_deploy_map/server/config_fed_server.conf:9]: This direction should be from executor back to controller.

File: job_templates/sag_pt_deploy_map/site-1/config_fed_client.conf [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          # if global_evaluation is true, the executor will expect the
+          # custom code to send back the evaluation metric
+          global_evaluation = true
+        }
+      }
+    }
+  ],
+
+  # this defined an array of task data filters. If provided, it will control the data to server
+  task_data_filters =  []
```
Comment (YuanTingHsieh) [inline job_templates/sag_pt_deploy_map/site-1/config_fed_client.conf:59]: it filters task data, the direction is from controller to executor.

**Pull request merged**